src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php

```diff
@@ -26,22 +28,43 @@ class AccessDecisionManager implements AccessDecisionManagerInterface
     const STRATEGY_CONSENSUS = 'consensus';
     const STRATEGY_UNANIMOUS = 'unanimous';
 
+    /**
+     * @var VoterInterface[]
+     */
     private $voters;
+
+    /**
+     * @var string
+     */
     private $strategy;
+
+    /**
+     * @var bool
+     */
     private $allowIfAllAbstainDecisions;
+
+    /**
+     * @var bool
+     */
     private $allowIfEqualGrantedDeniedDecisions;
 
     /**
+     * @var VoteReportBuilderInterface
+     */
+    private $reportBuilder;
+
+    /**
      * Constructor.
      *
-     * @param VoterInterface[] $voters                             An array of VoterInterface instances
-     * @param string           $strategy                           The vote strategy
-     * @param bool             $allowIfAllAbstainDecisions         Whether to grant access if all voters abstained or not
-     * @param bool             $allowIfEqualGrantedDeniedDecisions Whether to grant access if result are equals
+     * @param VoterInterface[]           $voters                             An array of VoterInterface instances
+     * @param VoteReportBuilderInterface $reportBuilder                      A vote report builder instance
+     * @param string                     $strategy                           The vote strategy
+     * @param bool                       $allowIfAllAbstainDecisions         Whether to grant access if all voters abstained or not
+     * @param bool                       $allowIfEqualGrantedDeniedDecisions Whether to grant access if result are equals
      *
      * @throws \InvalidArgumentException
      */
-    public function __construct(array $voters = array(), $strategy = self::STRATEGY_AFFIRMATIVE, $allowIfAllAbstainDecisions = false, $allowIfEqualGrantedDeniedDecisions = true)
+    public function __construct(array $voters = array(), VoteReportBuilderInterface $reportBuilder, $strategy = self::STRATEGY_AFFIRMATIVE, $allowIfAllAbstainDecisions = false, $allowIfEqualGrantedDeniedDecisions = true)
```

Comment:
it must be optional, and it must be at the end. Shifting all arguments is a BC break